features/lwipstack/LWIPStack.cpp

```diff
@@ -230,14 +230,28 @@ nsapi_error_t LWIP::socket_open(nsapi_socket_t *handle, nsapi_protocol_t proto)
         return NSAPI_ERROR_NO_SOCKET;
     }
 
-    enum netconn_type lwip_proto = proto == NSAPI_TCP ? NETCONN_TCP : NETCONN_UDP;
+    enum netconn_type netconntype;
+    if (proto == NSAPI_TCP) {
+        netconntype = NETCONN_TCP;
+    } else if (proto == NSAPI_UDP) {
+        netconntype = NETCONN_UDP;
+    } else if (proto == NSAPI_ICMP) {
+        netconntype = NETCONN_RAW;
+    } else {
+        return NSAPI_ERROR_PROTO_UNKNOWN;
```

Comment:
I think `NSAPI_ERROR_UNSUPPORTED` will do here, rather than adding an extra code. (@SeppoTakalo second opinion?)

Reply:
Yes, I agree. `NSAPI_ERROR_UNSUPPORTED` should be the error code.
I don't like too specific error codes as the enum starts to span too wide.